lib/documents.js

```diff
@@ -3338,6 +3338,280 @@ function copyConstraints(suggestConstraints, searchConstraints) {
+    }
+    jobState.stream = (inputKindValue === 'string') ? new stream.PassThrough():
+        new stream.PassThrough({objectMode: true});
+    jobState.inputKindValue = inputKindValue;
+  }
+  else {
+    jobState.stream = new stream.PassThrough({objectMode: true});
+  }
+
+  if(jobState.jobOptions.concurrentRequests) {
+    const concurrentRequests = jobState.jobOptions.concurrentRequests;
+    if (concurrentRequests.multipleOf) {
+      if (!new Set(['forests', 'hosts']).has(String(concurrentRequests.multipleOf))) {
+        throw new Error('Invalid value for multipleOf. Value must be forests or hosts.');
+      }
+    } else {
+      concurrentRequests.multipleOf = 'forests';
+    }
+
+    if (concurrentRequests.multiplier && concurrentRequests.multiplier <= 0) {
+      throw new Error('concurrentRequests.multiplier cannot be less than one');
+    } else {
+      concurrentRequests.multiplier = 8;
+    }
+    jobState.jobOptions.concurrentRequests = concurrentRequests;
+  }
+  else {
+    jobState.jobOptions.concurrentRequests = {multipleOf:'forests', multiplier:8};
+  }
+
+  if(jobState.jobOptions.batchSize){
+    if(jobState.jobOptions.inputKind === 'array'){
+      throw new Error('batchSize not expected when inputKind is array.');
+    }
+    if(jobState.jobOptions.batchSize <=0){
+      throw new Error('Invalid batchSize. batchSize cannot be less than or equal to 0.');
+    }
+    else if(jobState.jobOptions.batchSize > 100000){
+      throw new Error('Invalid batchSize. batchSize cannot be greater than 100000.');
+    }
+  } else {
+    jobState.jobOptions.batchSize = 100;
+  }
+
+  if(jobState.jobOptions.transformStrategy){
+    if (!new Set(['replace', 'ignore']).has(String(jobState.jobOptions.transformStrategy))) {
+      throw new Error('Invalid value for transformStrategy. Value must be replace or ignore.');
+    }
+  }
+
+  requester.startRequest(operation).result(mlutil.callbackOn(jobState, onTransformAllInit))
+      .catch((err) => {
+        jobState.stream.emit('error', err);
+        jobState.requesterCount = 0;
+        jobState.error = err;
+        finishTransformAll(jobState);
+      });
+
+  return jobState.stream;
+}
+
+function onTransformAllInit(output) {
+  const jobState = this;
+  let endpoint = '/v1/internal/apply-transform?start=1&pageLength='+jobState.jobOptions.batchSize;
+
+  if(jobState.jobOptions.transformStrategy){
+    endpoint = endpoint+'&result='+String(jobState.jobOptions.transformStrategy);
+  }
+
+  let connectionParams = jobState.docInstance.client.getConnectionParams();
+  let requestOptions = mlutil.copyProperties(connectionParams);
+  requestOptions.method = 'POST';
+  requestOptions.headers = {
+    'Accept':       'text/uri-list',
+    'Content-Type': 'application/json'
+  };
+  requestOptions.path = mlutil.databaseParam(connectionParams, jobState.endpoint);
+
+  let operation = new Operation(
+      'transform docs', jobState.docInstance.client, requestOptions, 'multipart', 'single'
```

Comment:
I believe both the request payload and response payloads should be 'single'

The URI list sent for transformation should be a single payload instead of sending multiple HTTP body payloads as parts.